useragent/user_agent.go

```diff
@@ -97,6 +85,24 @@ func (u info) String() string {
 
 type data []info
 
+// With always uses the latest value for a given alphanumeric key.
+// Panics if key or value don't satisfy alphanumeric or semver format.
+func (d data) With(key, value string) data {
+	if err := matchAlphanum(key); err != nil {
+		panic(err)
+	}
+	if err := matchAlphanumOrSemVer(value); err != nil {
+		panic(err)
+	}
+	for i := range d {
+		if d[i].Key == key {
+			d[i].Value = value
+			return d
```

Comment:
This should return a copy of the slice, not do an in-place modification.

As is, this will affect a parent context as well.